### PR TITLE
Collect separate statistics for failed tasks

### PR DESCRIPTION
**Pull request by losipiuk**
## Description

Collect separate statistics for failed tasks.
Show aggregated CPU/Scheduled time and cumulative memory for failed tasks in Web UI.

> Is this change a fix, improvement, new feature, refactoring, or other?

improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

query engine/UI

## Related issues, pull requests, and links

fixes: https://github.com/trinodb/trino/issues/10734
improves on: https://github.com/trinodb/trino/issues/10754

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# UI
* Add CPU time, scheduled time, and cumulative memory statistics regarding failed tasks in a query to web UI. ({issue}`10754`)

# General
* Add CPU time, scheduled time, and cumulative memory statistics regarding failed tasks in a query to query-completion event. ({issue}`10734`)

```

### CODE REVIEW

**Comment by losipiuk** [comment]
@arhimondr please skim if it looks like a good direction to you.

File: core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java [hunk 1]
```diff
@@ -72,9 +74,11 @@
     private final DataSize rawInputDataSize;
     private final long rawInputPositions;
     private final long cumulativeUserMemory;
+    private final long failedCumulativeUserMemory;
     private final DataSize userMemoryReservation;
     private final DataSize totalMemoryReservation;
     private final Duration totalCpuTime;
+    private final Duration failedCpuTime;
```
Comment (arhimondr) [inline core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java:81]: maybe "wasted" (here and in other places)?

Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java:81]: I personally like `failed` more as it is more precise and does not add emotional meaning to json field. But I do not feel super strongly. @martint / @findepi opinion?

Comment (findepi) [reply core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java:81]: i'd prefer "failed" too.

btw do  you plan anything like speculative execution, where work could become "wasted" without being "failed"?

Comment (findepi) [reply core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java:81]: eg "cancelled"

Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java:81]: Yeah - that will come up. But even then "wasted" is wrong term really. It has negative connotation, and if we are up to kill some tasks to speed up the execution of whole query it should not be seen as we are doing something wrong. And "wasting" resources feels wrong. So maybe some other term - neither "failed" nor "wasted" - but nothing comes to my mind.

File: core/trino-main/src/main/resources/webapp/src/components/QueryDetail.jsx [hunk 2]
```diff
@@ -1336,6 +1336,14 @@ export class QueryDetail extends React.Component {
...
+                                        <td className="info-title">
+                                            CPU Time (failed tasks)
+                                        </td>
+                                        <td className="info-text">
+                                            {query.queryStats.failedCpuTime}
+                                        </td>
+                                    </tr>
                                     <tr>
                                         <td className="info-title">
                                             Scheduled Time
```
Comment (arhimondr) [inline core/trino-main/src/main/resources/webapp/src/components/QueryDetail.jsx:1349]: nit: Scheduled time is also something we might want to have for failed tasks. I wonder if we should add it.

**Review by arhimondr — APPROVED** [review]

File: core/trino-main/src/main/resources/webapp/src/components/QueryDetail.jsx (hunk 2 above)
Comment (losipiuk) [reply core/trino-main/src/main/resources/webapp/src/components/QueryDetail.jsx:1349]: Makes sense. It is an omission - will add.

File: core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java (hunk 1 above)
Comment (martint) [reply core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java:82]: FWIW, I think "failed" is ok. The failure could be intrinsic (cluster / resource / system issues) or induced (task was forced to abort). In either case, it failed to complete.

**Pull request merged**